Source/WebCore/platform/KeyboardScrollingAnimator.cpp

```diff
@@ -277,6 +287,8 @@ bool KeyboardScrollingAnimator::beginKeyboardScrollGesture(const PlatformKeyboar
     m_idealPositionForMinimumTravel = m_scrollAnimator.currentPosition() + m_currentKeyboardScroll->offset;
     m_scrollController.willBeginKeyboardScrolling();
 
+    page->setCurrentKeyboardScrollingAnimator(this);
```

Comment:
You'll need to find a way to avoid this layering violation. Maybe the caller is the one that pushes the animator onto Page.